gr-fec/lib/reed-solomon/rstest.c

```diff
@@ -76,40 +65,27 @@ int main()
         printf("OK\n");
     }
 #endif
-
     for (i = 0; Tab[i].symsize != 0; i++) {
         int nn, kk;
 
         nn = (1 << Tab[i].symsize) - 1;
         kk = nn - Tab[i].nroots;
         printf("Testing (%d,%d) RS codec...", nn, kk);
         fflush(stdout);
-        if (Tab[i].symsize <= 8) {
-            if ((handle = init_rs_char(Tab[i].symsize,
-                                       Tab[i].genpoly,
-                                       Tab[i].fcs,
-                                       Tab[i].prim,
-                                       Tab[i].nroots)) == NULL) {
-                printf("init_rs_char failed!\n");
-                continue;
-            }
+        if ((handle = init_rs_char(Tab[i].symsize,
+                                    Tab[i].genpoly,
+                                    Tab[i].fcs,
+                                    Tab[i].prim,
+                                    Tab[i].nroots)) == NULL) {
+            printf("init_rs_char failed!\n");
+            continue;
+
             errs = exercise_char(handle, Tab[i].ntrials);
-        } else {
-#ifdef ALL_VERSIONS
-            if ((handle = init_rs_int(Tab[i].symsize,
-                                      Tab[i].genpoly,
-                                      Tab[i].fcs,
-                                      Tab[i].prim,
-                                      Tab[i].nroots)) == NULL) {
-                printf("init_rs_int failed!\n");
-                continue;
-            }
-            errs = exercise_int(handle, Tab[i].ntrials);
-#else
+        } 
+#if !defined(ALL_VERSIONS)
             printf("init_rs_init support is not enabled\n");
             exit(1);
 #endif
```

Comment:
What is the intent here?

Reply:
This piece of code wasn't really doing anything in the original code. I imagine that the intent was that if by accident a test case with `symsize > 8` was declared outside the `#ifdef ALL_VERSIONS`, then the test would print and error and finish, because `init_rs_int()`, which is necessary to test that test case, was only used when `ALL_VERSIONS` was defined. Now that we've removed the `_int()` functions, this piece of code is not only not doing anything, but also it doesn't play any role anymore, since now there should be no test cases with `symsize > 8` defined at all.

By the way, I realized that the original code uses `#ifdef ALL_VESIONS` in one place instead of `#ifdef ALL_VERSIONS`, but this doesn't matter because we've now deleted all this.